client/src/components/PetProfile.jsx

```diff
@@ -18,20 +18,17 @@ import { PetProfileTabs } from "./PetProfileTabs";
 import { useParams } from "react-router-dom";
 import { OWNER } from "../utilities/dummydata";
 
-export function PetProfile() {
+export function PetProfile({ baseURL }) {
```

Comment:
This is not great to be required to pass `baseURL` around everywhere. 

There are other solutions to consider:
1. have some sort of `import { baseURL } from ...` that you can use 
2. Look at `useContext`. It lets you distribute something like this to all the components. In my mind, this is a little overboard, as this value won't change throughout the running of your code.... but it is a solution that solve the problem of passing stuff around. 
3. Make your own version of `fetch`, let's say `fetchFromServer` (or something like that), that you use instead of the raw `fetch`. This is what I've done, as I've found I need to add my own headers (for auth, and other things). In this case, you'd have a `path` param or property, with an implicit server.

Reply:
I would love to understand solution 3 better--I tried useContext but it felt very tedious for the (relatively) small number of times I'm having to use the baseURL, and in the end I imported it as a prop just to get the app running. I will put this in our rolling agenda to look at on Monday.